### PR TITLE
DM-47517: Transfer the new warping tasks to drp_tasks

### PR DESCRIPTION
**Pull request by arunkannawadi**

### CODE REVIEW

File: python/lsst/pipe/tasks/coaddBase.py [hunk 1]
```diff
@@ -251,3 +252,36 @@ def subBBoxIter(bbox, subregionSize):
                                    "colShift=%s, rowShift=%s" %
                                    (bbox, subregionSize, colShift, rowShift))
             yield subBBox
+
+
+# Note that this is implemented as a free-floating function to enable reuse in
+# lsst.pipe.tasks.makeWarp and in lsst.drp.tasks.make_psf_matched_warp
+# without creating any relationships between the two classes.
+# This may be converted to a method after makeWarp.py is removed altogether.
```
Comment (enourbakhsh) [inline python/lsst/pipe/tasks/coaddBase.py:260]: I think you have the ticket number for removing `makeWarp.py`. I’d suggest adding it here.

File: python/lsst/pipe/tasks/makeWarp.py [hunk 2]
```diff
@@ -36,8 +36,7 @@
 from lsst.meas.algorithms import CoaddPsf, CoaddPsfConfig, GaussianPsfFactory
 from lsst.skymap import BaseSkyMap
 from lsst.utils.timer import timeMethod
-from .coaddBase import CoaddBaseTask, makeSkyInfo, reorderAndPadList
-from .make_psf_matched_warp import growValidPolygons
+from .coaddBase import CoaddBaseTask, makeSkyInfo, reorderAndPadList, growValidPolygons
```
Comment (enourbakhsh) [inline python/lsst/pipe/tasks/makeWarp.py:39]: nit:
```suggestion
from .coaddBase import CoaddBaseTask, growValidPolygons, makeSkyInfo, reorderAndPadList
```

File: python/lsst/pipe/tasks/make_direct_warp.py [hunk 3]
```diff
@@ -19,956 +19,19 @@
...
-    doRevertOldBackground = Field[bool](
-        doc="Revert the old backgrounds from the `background_revert_list` "
-            "connection?",
-        default=False,
+# TODO: Remove this entire file in DM-47521.
+
+import warnings
+
+try:
+    from lsst.drp.tasks.make_direct_warp import *  # noqa: F401, F403
```
Comment (enourbakhsh) [inline python/lsst/pipe/tasks/make_direct_warp.py:27]: I assume this wildcard import is safe. Same in `python/lsst/pipe/tasks/make_psf_matched_warp.py`.

**Review by enourbakhsh — APPROVED** [review]
Left a few minor comments. Marking this as `approved`.

File: python/lsst/pipe/tasks/make_direct_warp.py (hunk 3 above)
Comment (arunkannawadi) [reply python/lsst/pipe/tasks/make_direct_warp.py:27]: Well, given that it is meant to mimic the one in `drp_tasks`, it is. But I'll make it limited to what was originally present at the time of this transfer.

**Pull request merged**